lib/collection/proxy-config.js

```diff
@@ -118,13 +122,42 @@ _.assign(ProxyConfig.prototype, /** @lends ProxyConfig.prototype */ {
         }
 
         _.isString(options.match) && (this.match = new UrlMatchPattern(options.match));
+        _.isString(_.get(options, 'match.pattern')) && (this.match = new UrlMatchPattern(options.match.pattern));
```

Comment:
this case was missed in previous PR
This is when in the update we pass an instance of proxyConfig
we should support both 
```js
{ match: 'some string' }
```
and
```js
{ match:  {pattern : 'some string' } }
```